maya/core.py

```diff
@@ -43,6 +43,39 @@ def wrapper(self, *args, **kwargs):
     return inner
 
 
+def validate_arguments_type_of_function(param_type=None):
+    """
+    Decorator to validate the <type> of arguments in
+    the calling function are of the `param_type` class.
+
+    if `param_type` is None, uses `param_type` as the class where it is used.
+
+    Note: Use this decorator on the functions of the class.
+    """
+    inner_dict = {   # To use arguments within the inner functions
```

Comment:
Oh, I've missed the assignment later in the code. You might want to do something like `type_ = param_type or type(self)` and use `type_` to avoid this extra dict.